core/unit_test/TestConcepts.hpp

```diff
@@ -59,4 +59,156 @@ static_assert(std::is_same<int, Kokkos::Impl::remove_cvref_t<int &>>{}, "");
+  using LeagueRankArchetypeExpr =
+      decltype(std::declval<U const &>().league_rank());
+
+  template <class U>
+  using LeagueSizeArchetypeExpr =
+      decltype(std::declval<U const &>().league_size());
+
+  // scratch space
+  template <class U>
+  using TeamShmemArchetypeExpr =
+      decltype(std::declval<U const &>().team_shmem());
+
+  template <class U>
+  using TeamScratchArchetypeExpr =
+      decltype(std::declval<U const &>().team_scratch(0));
+
+  template <class U>
+  using ThreadScracthArchetypeExpr =
+      decltype(std::declval<U const &>().thread_scratch(0));
+
+  // collectives
+  template <class U>
+  using TeamBarrierArchetypeExpr =
+      decltype(std::declval<U const &>().team_barrier());
+
+  template <class U>
+  using TeamBroadcastArchetypeExpr = decltype(
+      std::declval<U const &>().team_broadcast(lvalueForMethodsNeedingIt_, 0));
+
+  template <class U>
+  using TeamBroadcastAcceptClosureArchetypeExpr =
+      decltype(std::declval<U const &>().team_broadcast(
+          TrivialFunctor{}, lvalueForMethodsNeedingIt_, 0));
+
+  template <class U>
+  using TeamReducedArchetypeExpr =
+      decltype(std::declval<U const &>().team_reduce(
+          std::declval<reduction_to_test_t>()));
+
+  template <class U>
+  using TeamScanArchetypeExpr = decltype(std::declval<U const &>().team_scan(
+      lvalueForMethodsNeedingIt_, ptrForMethodsNeedingIt_));
+
+ public:
+  static constexpr bool value =
+      Kokkos::is_detected_v<ExecutionSpaceArchetypeAlias, T> &&
+      Kokkos::is_detected_v<ScratchMemorySpaceArchetypeAlias, T> &&
+      //
+      Kokkos::is_detected_exact_v<int, TeamRankArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<int, TeamSizeArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<int, LeagueRankArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<int, LeagueSizeArchetypeExpr, T> &&
+      //
+      Kokkos::is_detected_exact_v<
+          Kokkos::detected_t<ScratchMemorySpaceArchetypeAlias, T> const &,
+          TeamShmemArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<
+          Kokkos::detected_t<ScratchMemorySpaceArchetypeAlias, T> const &,
+          TeamScratchArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<
+          Kokkos::detected_t<ScratchMemorySpaceArchetypeAlias, T> const &,
+          ThreadScracthArchetypeExpr, T> &&
+      //
+      Kokkos::is_detected_exact_v<void, TeamBarrierArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<void, TeamBroadcastArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<void, TeamBroadcastAcceptClosureArchetypeExpr,
+                                  T> &&
+      Kokkos::is_detected_exact_v<void, TeamReducedArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<test_value_type, TeamScanArchetypeExpr, T>;
+  constexpr operator bool() const noexcept { return value; }
+};
+
+template <class T>
+inline constexpr bool is_team_handle_complete_trait_check_v =
+    is_team_handle_complete_trait_check<T>::value;
```

Comment:
```suggestion
    is_team_handle_complete_trait_check<T>::value && Kokkos::is_team_handle_v<T>;
```
that way you don't need to duplicate all that code below

Reply:
I think its fine. Makes it more obvious that the primary team handle is tested.